### ISSUE TITLE
RCS1229 false positive with ValueTask

### ISSUE THREAD

**Issue by antoinebj** [dotnet/roslynator#1617]
Similar to #821 but for `ValueTask`

The following code triggers RCS1229, which disappears if the return type is changed to `Task`:
```csharp
public static class Rcs1229Repro
{
    public static ValueTask Foo()
    {
        using var memoryStream = new MemoryStream();

        return ValueTask.CompletedTask;
    }
}
```

(btw there's a typo in the method name [here](https://github.com/dotnet/roslynator/blob/main/src/Analyzers/CSharp/Analysis/UseAsyncAwaitAnalyzer.cs#L310C29-L310C39), there is a second "Or" instead of "Of" in `IsTaskOrTaskOrT`)
